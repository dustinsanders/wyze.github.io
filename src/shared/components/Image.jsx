// @flow

import type { Style, VNode } from '../../types'
import { css } from 'glamor'
import preact from 'preact'

export type Props = {
  alt?: string,
  circle?: boolean,
  ext: string,
  name: string,
  width?: number,
}

const maxHeight: Style = css({
  maxHeight: 100,
})
const round: Style = css({
  borderRadius: '50%',
})

const Image = ({ alt, circle, ext, name, width }: Props): VNode => (
  <img
    {...css(maxHeight, circle && round)}
    alt={alt || name}
    height="100"
    src={`img/${name}.${ext}`}
    width={width}
  />
)

Image.defaultProps = {
  alt: '',
  circle: false,
  width: 100,
}

export default Image
