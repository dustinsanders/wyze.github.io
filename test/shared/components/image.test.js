// @flow

import Image from '../../../src/shared/components/Image'
import snapshot from '../../helpers/snapshot'

const image = {
  ext: 'jpg',
  name: 'me',
}

const imageAsCircle = {
  ...image,
  circle: true,
}

const imageWithAlt = {
  ...image,
  alt: 'Neil Kistner',
}

const imageWithWidth = {
  ...image,
  width: 200,
}

describe('<Image />', () => {
  it('renders', snapshot.bind(null, Image, image))
  it('renders as circle', snapshot.bind(null, Image, imageAsCircle))
  it('renders custom alt attr', snapshot.bind(null, Image, imageWithAlt))
  it('renders custom width', snapshot.bind(null, Image, imageWithWidth))
})
