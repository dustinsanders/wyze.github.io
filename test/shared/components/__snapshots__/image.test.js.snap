// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Image /> renders 1`] = `"<img data-css-1o7jcq0 alt=\\"me\\" height=\\"100\\" src=\\"img/me.jpg\\" width=\\"100\\" />"`;

exports[`<Image /> renders as circle 1`] = `"<img data-css-p1gv9z alt=\\"me\\" height=\\"100\\" src=\\"img/me.jpg\\" width=\\"100\\" />"`;

exports[`<Image /> renders custom alt attr 1`] = `"<img data-css-1o7jcq0 alt=\\"Neil Kistner\\" height=\\"100\\" src=\\"img/me.jpg\\" width=\\"100\\" />"`;

exports[`<Image /> renders custom width 1`] = `"<img data-css-1o7jcq0 alt=\\"me\\" height=\\"100\\" src=\\"img/me.jpg\\" width=\\"200\\" />"`;
